compiler/resolution/functionResolution.cpp

```diff
@@ -8165,6 +8225,9 @@ static void resolveAutoCopies() {
         ts->hasFlag(FLAG_SYNTACTIC_DISTRIBUTION) == false) {
       if (AggregateType* at = toAggregateType(ts->type)) {
         if (isRecord(at) == true) {
+          // If we attempt to resolve auto-copy and co. for an infinite record
+          // we may enter an infinite loop and the compiler will crash.
+          checkForInfiniteRecord(at);
```

Comment:
Please comment in the PR message why checkForInfiniteRecord() is in such odd places like wrapAndCleanUpActuals() and resolveAutoCopies(). Seems like it should occur when you are looking at the record itself.

Reply:
Resolution doesn't always give us a chance to properly 'handle' a record before we attempt to use its fields, so I inserted these checks right before we would otherwise encounter a segfault.

Reply:
Thanks for updating the PR text.